analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -43,9 +44,77 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         {
             state = state.ResetFieldConstraints();
         }
-        return invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace))
-            ? ProcessStringIsNullOrEmpty(context, invocation)
-            : new[] { state };
+        if (invocation.TargetMethod.ContainingType.Is(KnownType.System_Linq_Enumerable))
+        {
+            return ProcessLinqEnumerable(context, invocation);
+        }
+        else if (invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)))
+        {
+            return ProcessStringIsNullOrEmpty(context, invocation);
+        }
+        else
+        {
+            return new[] { state };
+        }
+    }
+
+    private static ProgramState[] ProcessLinqEnumerable(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        switch (invocation.TargetMethod.Name)
+        {
+            case "Append":
+            case nameof(Enumerable.AsEnumerable):
+            case nameof(Enumerable.Cast):
+            case "Chunk":
+            case nameof(Enumerable.Concat):
+            case nameof(Enumerable.Distinct):
```

Comment:
DefaultIfEmpty (never returns null. Just guarantees, that the collection is not empty)